src/api/rolesApi.tsx

```diff
@@ -105,14 +119,36 @@ export const useAddRole = () => {
 
 export const useDeleteRole = () => {
   const queryClient = useQueryClient();
+  const roles = useRoles();
   const { dispatchToast } = useToastController("toaster");
   return useMutation(
     ["deleteRole"],
     async (id: number) => {
-      return spWebContext.web.lists
+      let error = undefined;
+      const thisRole = roles.data?.find((role) => (role.Id = id));
+
+      let result = await spWebContext.web.lists
```

Comment:
result is never reassigned
```suggestion
      const result = await spWebContext.web.lists
```